x/timerstore/README.md

```diff
@@ -0,0 +1,87 @@
+
+The timerstore allows other modules to create timers, that triggers a callback functions.  
+A timer defined in this module can be of two types: BlockHeight or BlockTime.  
+The callback function can be triggered either at BeginBlock or EndBlock.
+
+## Contents
+
+- [Concepts](#concepts)
+  - [TimerStore](#timerstore)
+  - [BeginBlock & EndBlock](#beginblock--endblock)
+  - [BlockHeight & BlockTime](#blockheight--blocktime)
+  - [Trigger](#trigger)
+- [Parameters](#parameters)
+- [Queries](#queries)
+- [Transactions](#transactions)
+- [Proposals](#proposals)
+
+## Concepts
+
+### TimerStore
+
+The `TimerStore` object is the main entity of this module, and holds all of its logic.
+The `timerstore` module's keeper is pretty straight forward:
+
+```go
+type Keeper struct {
+	timerStoresBegin []*timerstoretypes.TimerStore
+	timerStoresEnd   []*timerstoretypes.TimerStore
+	cdc              codec.BinaryCodec
+}
+```
+
+Whenever a module creates a new timer, it will be stored in `timerStoresBegin` or in `timerStoresEnd`, depending on the function that was used to create the timer.
+
+### BeginBlock & EndBlock
+
+A module can decide whether to trigger a certain timer at the start of the block or at the end of it.  
+The function `NewTimerStoreBeginBlock` will create a new `TimerStore` that will trigger at the `BeginBlock` when the time is right, and the function `NewTimerStoreEndBlock` will create a new `TimerStore` that will trigger at the `EndBlock` when the time is right.
+
+### BlockHeight & BlockTime
+
+After calling the function `NewTimerStoreBeginBlock` or `NewTimerStoreEndBlock` the calling procedure can use the returned `TimerStore` object to set the callback of the timer. The callback has to be of the signature `func(ctx sdk.Context, key, data []byte)`, more on that in the next section.
+
+To set the callback, the procedure needs to call `WithCallbackByBlockHeight` or `WithCallbackByBlockTime`.
+The `WithCallbackByBlockHeight` function will add the callback to the `TimerStore` object, under the `BlockHeight` kind, and the `WithCallbackByBlockTime` function will add the callback to the `TimerStore` object, under the `BlockTime` kind.
+
+The `BlockHeight` callbacks, can be triggered at a given block, and the `BlockTime` callback can be triggered at the first block that it's time is past the given time.
+
+### Trigger
+
+As stated in the previous section, timers can be one of two kinds: `BlockHeight` or `BlockTime`.  
+Once a callback is created in the TimerStore, a procedure can then create a new timer corresponding to its kind.
+
+For `BlockHeight` callbacks, the function `AddTimerByBlockHeight(ctx sdk.Context, block uint64, key, data []byte)` will create a timer that will trigger at `block`, with the given `key` and `data`.
```

Comment:
emphasize the the `key` argument is actually the timer's subkey, so it doesn't have to be unique (the timers' primary key is unique and is created inside the timerstore module). Also, note that the `data` argument is useful for attaching data to the timer that will be available when the callback function is triggered

Reply:
Clarified in 620343f3452b5248733823547ecdc6d7fc62010c